pkg/database/entities/expenses_summary.go

```diff
@@ -0,0 +1,72 @@
+package entities
+
+import (
+	"ExpensesTracker/pkg/api/http/responses"
+	"time"
+)
+
+type ExpensesSummary struct {
+	Id                        uint
+	Name                      string `gorm:"type:character varying(200)"`
+	CreatedAt                 time.Time
+	UpdatedAt                 time.Time
+	ClosedAt                  time.Time `gorm:"type:timestamp with time zone;default:null"`
+	UsdToPlnRatio             float64   `gorm:"type:numeric(4,2)"`
+	MoneyTransferredToSavings float64   `gorm:"type:numeric(10,2)"`
+	Incomes                   []Income
+	Expenses                  []Expense
+	Savings                   []Saving
+}
+
+func (expensesSummary *ExpensesSummary) MapToCreateExpensesSummaryResponse() *responses.CreateExpensesSummaryResponse {
+	var closedAt *time.Time
+	if expensesSummary.ClosedAt.IsZero() {
+		closedAt = nil
+	}
+
+	expensesResponse := responses.CreateExpensesSummaryResponse{
+		Id:                        expensesSummary.Id,
+		Name:                      expensesSummary.Name,
+		CreatedAt:                 expensesSummary.CreatedAt,
+		UpdatedAt:                 expensesSummary.UpdatedAt,
+		ClosedAt:                  closedAt,
+		UsdToPlnRatio:             expensesSummary.UsdToPlnRatio,
+		MoneyTransferredToSavings: expensesSummary.MoneyTransferredToSavings,
+	}
+
+	return &expensesResponse
+}
+
+func (expensesSummary *ExpensesSummary) MapToGetExpensesSummaryResponse() *responses.GetExpensesSummaryResponse {
+	var closedAt *time.Time
+	if expensesSummary.ClosedAt.IsZero() {
+		closedAt = nil
+	}
+
+	expensesResponse := responses.GetExpensesSummaryResponse{
+		Id:                        expensesSummary.Id,
+		Name:                      expensesSummary.Name,
+		CreatedAt:                 expensesSummary.CreatedAt,
+		UpdatedAt:                 expensesSummary.UpdatedAt,
+		ClosedAt:                  closedAt,
+		UsdToPlnRatio:             expensesSummary.UsdToPlnRatio,
+		MoneyTransferredToSavings: expensesSummary.MoneyTransferredToSavings,
+	}
+
+	expensesResponse.Incomes = make(map[string]float32)
+	for _, element := range expensesSummary.Incomes {
+		expensesResponse.Incomes[element.Name] = element.Value
+	}
+
+	expensesResponse.Savings = make(map[string]float32)
+	for _, element := range expensesSummary.Savings {
+		expensesResponse.Savings[element.Name] = element.Value
+	}
+
+	expensesResponse.Expenses = make(map[string]float32)
+	for _, element := range expensesSummary.Expenses {
+		expensesResponse.Expenses[element.Name] = element.Value
+	}
```

Comment:
I am not sure if this mapping could have been done any better :/ Its simple yet its not exactly a DRY rule though

Reply:
Maybe by creating Interface 'IMappable' with function returning pointer to map[string]float32?

Reply:
On second thought I am not entirely sure if it that will work. Even if I mark them with an interface I can't define a method that could envoked on `[]Income` (not `Income`) - similarly `Saving` and `Expense`. Unfortunately its not supported in Go (I think) which is a pity. First thing that is somewhat similar to this problem that comes to my mind is append function but its not a method, its a simple function with global scope of visibility `func append(slice []Type, elems ...Type) []Type`. Given that I will probably stay with the current solution

Reply:
Another option for interface would be marking `Income`, `Saving` and `Expense` with an interface for common properties but... its also not supported in Go. Only methods

Reply:
I think we should leave it as it is then. Go is not enough objective language to resolve it with abstraction layer